### PR TITLE
Preparing for Dart 2.8 release

### PR DESCRIPTION
**Pull request by domesticmouse**
@kwalrath How do you feel about Gitpod.io? I'm using it as an easy way to update Stagehand under the pre-release version of Dart.

@mit-mit I tried using `pub outdated`, but hit an interesting issue with our Angular sample. By my reading of the tool output I should be updating dependency versions, but these versions are the maximum before we hit errors. We should probably have `pub outdated` figure out what the maximum allowed version is and either suggest an edit, or stay silent when upgrades are forbidden.

```bash
Checking /workspace/stagehand/templates/web-angular
Dependencies                Current    Upgradable  Resolvable  Latest    
all up-to-date              

dev_dependencies           
angular_test                2.3.0      2.3.0       2.4.0       2.4.0     
test                        1.6.3      1.6.3       1.14.2      1.14.2    

transitive dependencies    
analyzer                    0.35.4     0.35.4      0.37.1+1    0.39.8    
angular_ast                 0.5.9      0.5.9       0.5.11      0.5.11    
angular_compiler            0.4.3      0.4.3       0.4.5       0.4.5     
angular_forms               2.1.2      2.1.2       2.1.4       2.1.4     
csslib                      0.14.6+1   0.14.6+1    0.16.1      0.16.1    
dart_style                  1.2.4      1.2.4       1.2.10      1.3.6     
front_end                   0.1.14     0.1.14      0.1.21+1    0.1.29    
intl                        0.15.8     0.15.8      0.15.8      0.16.1    
kernel                      0.3.14     0.3.14      0.3.21+1    0.3.29    
matcher                     0.12.5     0.12.5      0.12.6      0.12.6    
protobuf                    0.13.16+1  0.13.16+1   0.13.16+1   1.0.1     
source_gen                  0.9.4+4    0.9.4+4     0.9.4+6     0.9.5     

transitive dev_dependencies
bazel_worker                0.1.21     0.1.21      0.1.21      0.1.23+1  
boolean_selector            1.0.5      1.0.5       2.0.0       2.0.0     
build_resolvers             1.2.1      1.2.1       1.2.1       1.3.7     
coverage                    -          -           0.13.9      0.13.9    
shelf_packages_handler      1.0.4      1.0.4       2.0.0       2.0.0     
source_map_stack_trace      1.1.5      1.1.5       2.0.0       2.0.0     
test_api                    0.2.5      0.2.5       0.2.15      0.2.15    
test_core                   0.2.5      0.2.5       0.3.3       0.3.3     
vm_service                  -          -           4.0.2       4.0.2     
webkit_inspection_protocol  -          -           0.5.0+1     0.5.0+1   

17  dependencies are constrained to versions that are older than a resolvable version.
To update these dependencies, edit pubspec.yaml.
```

### CODE REVIEW

**Comment by mit-mit** [comment]
>  I should be updating dependency versions, but these versions are the maximum before we hit errors. We should probably have pub outdated figure out what the maximum allowed version is

The tool (theoretically, at least) already does this; it's the resolvable column (i.e., you should be able to move to `>=` those listed versions).

**Comment by mit-mit** [comment]
> How do you feel about Gitpod.io? I'm using it as an easy way to update Stagehand under the pre-release version of Dart.

I'm really curious about this. Can you briefly explain what you would do without gitpod, and what the gitpod flow is?

**Comment by domesticmouse** [comment]
> > I should be updating dependency versions, but these versions are the maximum before we hit errors. We should probably have pub outdated figure out what the maximum allowed version is
> 
> The tool (theoretically, at least) already does this; it's the resolvable column (i.e., you should be able to move to `>=` those listed versions).

When I attempted to update `angular_test: ^2.4.0` it all fell apart. The same with uplifting `test` above 1.6.3.

**Comment by domesticmouse** [comment]
> > How do you feel about Gitpod.io? I'm using it as an easy way to update Stagehand under the pre-release version of Dart.
> 
> I'm really curious about this. Can you briefly explain what you would do without gitpod, and what the gitpod flow is?

I'm using gitpod as a way of running dart versions that aren't my installed (stable) version. For example, I use gitpod to maintain PRs like https://github.com/dart-lang/dart-services/pull/518 to bump dart revisions each time a new beta/dev dart release happens.

**Comment by mit-mit** [comment]
Odd, created https://github.com/dart-lang/pub/issues/2468

**Comment by kwalrath** [comment]
> @kwalrath How do you feel about Gitpod.io? I'm using it as an easy way to update Stagehand under the pre-release version of Dart.

I'm fine with it if people like @kevmoo and @mit-mit are fine with it.

**Comment by domesticmouse** [comment]
Hey @kevmoo, can I have the commit bit back on this repo? I'd like to assign you as a reviewer for this PR, along with @mit-mit and @kwalrath =)

File: .travis.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ dart:
   # SDK version to prevent errors when using Stagehand from inside VSCode's
   # Dart and Flutter plugins. For more detail, please see:
   # https://github.com/dart-lang/stagehand/issues/617
-  - 2.7.0
-  - dev
```
Comment (kevmoo) [inline .travis.yml:13]: we WANT `dev` – it helps us catch breaks in the SDK!

Comment (mit-mit) [reply .travis.yml:13]: Actually, isn't beta now better for that?

Comment (kevmoo) [reply .travis.yml:13]: dev is rolled more, so it helps us catch things early.

beta would also be nice, it just starts to slow things down aa bit...

Comment (mit-mit) [reply .travis.yml:13]: Yes, but dev also has false negatives, and I'm not convinced the once a month cadence for beta is a real issue.

**Comment by kevmoo** [comment]
The false negatives are bugs we can file and get fixed before we get to
beta!

**Comment by kevmoo** [comment]
...we've found/fixed A LOT of bugs this way!

File: .travis.yml (hunk 1 above)
Comment (domesticmouse) [reply .travis.yml:13]: Is Travis for stagehand run in a cronjob somewhere, or is it only run when I do the release cycle update?

File: templates/console-full/pubspec.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ description: A sample command-line application.
 # homepage: https://www.example.com
 
 environment:
-  sdk: '>=2.7.0 <3.0.0'
+  sdk: '>=2.8.0-20.11.beta <3.0.0'
 
 #dependencies:
 #  path: ^1.6.0
```
Comment (kwalrath) [inline templates/console-full/pubspec.yaml:10]:
```suggestion
#  path: ^1.7.0
```

(I just happened to notice that the current version is 1.7.0.)

File: templates/console-simple/pubspec.yaml [hunk 3]
```diff
@@ -4,7 +4,7 @@ description: A simple command-line application.
 # homepage: https://www.example.com
 
 environment:
-  sdk: '>=2.7.0 <3.0.0'
+  sdk: '>=2.8.0-20.11.beta <3.0.0'
 
 #dependencies:
 #  path: ^1.6.0
```
Comment (kwalrath) [inline templates/console-simple/pubspec.yaml:10]:
```suggestion
#  path: ^1.7.0
```

File: templates/package-simple/pubspec.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@ description: A starting point for Dart libraries or applications.
 # homepage: https://www.example.com
 
 environment:
-  sdk: '>=2.7.0 <3.0.0'
+  sdk: '>=2.8.0-20.11.beta <3.0.0'
 
 #dependencies:
 #  path: ^1.6.0
```
Comment (kwalrath) [inline templates/package-simple/pubspec.yaml:10]:
```suggestion
#  path: ^1.7.0
```

Comment (kwalrath) [reply templates/package-simple/pubspec.yaml:10]: I guess we suggest this everywhere... :) So I won't mark it anywhere else.

File: templates/console-full/pubspec.yaml (hunk 2 above)
Comment (domesticmouse) [reply templates/console-full/pubspec.yaml]: Done

File: templates/console-simple/pubspec.yaml (hunk 3 above)
Comment (domesticmouse) [reply templates/console-simple/pubspec.yaml]: done

File: templates/package-simple/pubspec.yaml (hunk 4 above)
Comment (domesticmouse) [reply templates/package-simple/pubspec.yaml]: done

File: .travis.yml (hunk 1 above)
Comment (domesticmouse) [reply .travis.yml:13]: Brought back `dev` and introduced `beta`. The `"2.8.0-20.11.beta"` entry will turn into whichever version 2.8 releases as.

**Comment by domesticmouse** [comment]
PTAL @kevmoo @mit-mit

**Comment by domesticmouse** [comment]
@mit-mit I’ll update this PR when @athomas pushes 2.8.1. That way we know the PR is green. I can leave submitting the PR and publishing the updated version to pub to @kwalrath so it happens during the official launch window, when I’ll be sound asleep.

**Comment by domesticmouse** [comment]
@mit-mit I believe everything is now in place for @kwalrath to land this PR and publish it. PTAL

**Pull request merged**